### ISSUE TITLE
ICosmosDBSerializerFactory not found for isolated function using cosmosdb trigger

### ISSUE THREAD

**Issue by bexxx** [Azure/azure-webjobs-sdk-extensions#818]
Adding a cusomt serializer is not working when following docs (isolated function, net7)

#### Repro steps

Provide the steps required to reproduce the problem

I try to run an azure function in isolated mode to switch to .net 7.
Using cosmosdb extension Microsoft.Azure.Functions.Worker.Extensions.CosmosDB", version 4.0.0 and  Microsoft.Azure.Cosmos" Version="3.31.2".
The trigger fails in deserialization to POCO type, so I'd like to use our System.Text.Json based serializer settings by providing the cosmosdb serializer factory.
I can't seem to find where (which nuget) the type ICosmosDBSerializerFactory is coming from. Any pointer is appreciated.

2. Step B

#### Expected behavior

The referenced nuget for cosmosdb extensions contains the type, or doc points them out.

#### Actual behavior

Compilation error because of missing dependencies

#### Known workarounds

Provide a description of any known workarounds.

#### Related information 

Provide any related information 

* Package version
* Links to source

**Comment by bexxx** [comment]
My current workaround is to use
IReadOnlyList<JsonNode> documents
and then do a Deseriazation of the document.ToString().

Would love to be able to properly implement it.

**Comment by ealsur** [comment]
@fabiocav / @kshyju Is this supported on Worker extensions?

**Comment by fabiocav** [comment]
@bexxx that extensibility point is not available in the Isolated model as the payload serialization is handled by the actual extension, running in the host process. 

We're actively working on enhancements to the binding model for all extensions, Cosmos DB included, and you can track this work [here](https://github.com/Azure/azure-functions-dotnet-worker/issues/1081). This will bring additional flexibility on the types you have the ability to bind against and better control over the binding behavior.

I would, however, like to better understand your scenario and requirement for custom serialization. Would you be able to expand a bit on what your custom serializer is doing or, better yet, share an example of what you're trying to do so we can take a closer look? There might be some options available today with Isolated depending on what the requirements are.

Thanks!

**Comment by bexxx** [comment]
We implemented our service in .net core 6/7 using System.Text.Json. Our serialization options are defined for each layer (API, Cosmosdb, internal service dependencies) to be in full control of the data format (they are different per layer). We do not have a lot of special cases, the biggest one is likely using enums as strings. In our implementation, we also do not use output bindings and write to cosmosdb from code directly.

When NOT adding any specific serialization code, I got exceptions while deserialization the cosmosdb entities while calling our trigger, because the enum type could not be deserialized by default. To get a step further, I could start to specify a duplicate set of serializer settings for newtonsoft, but this leaves a code smell because both settings could actually diverge (not likey, but also not clean). If this is helpful, I can check what changes are actually needed to get it running.

When I found the video from Matias, I was pretty happy that system.text.json is now nicely supported and I could just reuse the existing settings however the rest of the stack is implemented. This looked like a clean way of separation. I was not aware that both triggers (hosted, isolated) are so different and do not provide the same functionality. This was also not really called out in the docs about isolated functions. We started to switch to isolated, because we wanted to move to .net 7 which is not supported on hosted (all to be able to consume latest nugets to get latest security fixes).
